api/nodes/views.py

```diff
@@ -296,37 +295,23 @@ def bulk_destroy_skip_uneditable(self, resource_object_list, user, object_type):
 
     # Overrides BulkDestroyModelMixin
     def perform_bulk_destroy(self, resource_object_list):
-
-        auth = get_user_auth(self.request)
-        date = timezone.now()
-        id_list = [x.id for x in resource_object_list]
-
-        if NodeRelation.objects.filter(
-            parent__in=resource_object_list,
-            child__is_deleted=False,
-        ).exclude(Q(child__in=resource_object_list) | Q(is_node_link=True)).exists():
-            raise ValidationError('Any child components must be deleted prior to deleting this project.')
-
-        remove_addons(auth, resource_object_list)
+        if enforce_no_children(self.request):
+            if NodeRelation.objects.filter(
+                parent__in=resource_object_list,
+                child__is_deleted=False,
+            ).exclude(Q(child__in=resource_object_list) | Q(is_node_link=True)).exists():
+                raise ValidationError('Any child components must be deleted prior to deleting this project.')
 
         for node in resource_object_list:
-            node.add_remove_node_log(auth=auth, date=date)
-
-        nodes = AbstractNode.objects.filter(id__in=id_list)
-        nodes.update(is_deleted=True, deleted_date=date)
-        if nodes.filter(is_public=True).exists():
-            AbstractNode.bulk_update_search(resource_object_list)
-        for node in nodes:
-            project_signals.node_deleted.send(node)
+            self.perform_destroy(node)
 
-    # Overrides BulkDestroyJSONAPIView
+    # Overrides BulkDestroyModelMixin
     def perform_destroy(self, instance):
         auth = get_user_auth(self.request)
         try:
             instance.remove_node(auth=auth)
-        except NodeStateError as err:
-            raise ValidationError(str(err))
-        instance.save()
+        except PermissionsError as err:
+            raise PermissionDenied(str(err))
```

Comment:
Remove_node should no longer call a NodeStateError any longer, but a PermissionsError can be thrown if the user doesn't have perms to any node in the hierarchy.   No longer saving here because `remove_node` covers this.  We need to save within that function so deleted public nodes are removed from search.